File: generate-icons.js
const fs = require('fs');
const { createCanvas, loadImage } = require('canvas');

// Create a canvas for the icon
const canvas = createCanvas(1024, 1024);
const ctx = canvas.getContext('2d');

// Background gradient (white to light gray)
const gradient = ctx.createLinearGradient(0, 0, 1024, 1024);
gradient.addColorStop(0, '#FFFFFF');
gradient.addColorStop(1, '#F8F9FA');

// Draw background
ctx.fillStyle = gradient;
ctx.fillRect(0, 0, 1024, 1024);

// Draw chart container background
ctx.fillStyle = '#FFFFFF';
ctx.strokeStyle = '#E5E7EB';
ctx.lineWidth = 4;
ctx.beginPath();
ctx.roundRect(150, 200, 724, 500, 20);
ctx.fill();
ctx.stroke();

// Draw chart bars
const barData = [
    { x: 220, y: 350, height: 280 },
    { x: 330, y: 280, height: 350 },
    { x: 440, y: 400, height: 230 },
    { x: 550, y: 320, height: 310 },
    { x: 660, y: 250, height: 380 }
];

ctx.fillStyle = '#1F2937';
barData.forEach(bar => {
    ctx.beginPath();
    ctx.roundRect(bar.x, bar.y, 80, bar.height, 8);
    ctx.fill();
});

// Draw trend line
ctx.strokeStyle = '#3B82F6';
ctx.lineWidth = 6;
ctx.lineCap = 'round';
ctx.lineJoin = 'round';
ctx.beginPath();
ctx.moveTo(260, 350);
ctx.lineTo(370, 280);
ctx.lineTo(480, 400);
ctx.lineTo(590, 320);
ctx.lineTo(700, 250);
ctx.stroke();

// Draw trend dots
ctx.fillStyle = '#3B82F6';
const dots = [
    { x: 260, y: 350 },
    { x: 370, y: 280 },
    { x: 480, y: 400 },
    { x: 590, y: 320 },
    { x: 700, y: 250 }
];

dots.forEach(dot => {
    ctx.beginPath();
    ctx.arc(dot.x, dot.y, 8, 0, 2 * Math.PI);
    ctx.fill();
});

// Draw axes
ctx.strokeStyle = '#9CA3AF';
ctx.lineWidth = 2;
// X-axis
ctx.beginPath();
ctx.moveTo(200, 650);
ctx.lineTo(820, 650);
ctx.stroke();
// Y-axis
ctx.beginPath();
ctx.moveTo(200, 230);
ctx.lineTo(200, 650);
ctx.stroke();

// Draw "SoroOne" text
ctx.fillStyle = '#1F2937';
ctx.font = 'bold 120px Arial, sans-serif';
ctx.textAlign = 'center';
ctx.textBaseline = 'middle';

// Add text shadow
ctx.shadowColor = 'rgba(0, 0, 0, 0.15)';
ctx.shadowBlur = 8;
ctx.shadowOffsetX = 0;
ctx.shadowOffsetY = 4;

ctx.fillText('SoroOne', 512, 800);

// Save as PNG
const buffer = canvas.toBuffer('image/png');
fs.writeFileSync('./assets/icon.png', buffer);

// Create adaptive icon (simpler chart version)
const adaptiveCanvas = createCanvas(1024, 1024);
const adaptiveCtx = adaptiveCanvas.getContext('2d');

// White circle background
adaptiveCtx.fillStyle = '#FFFFFF';
adaptiveCtx.beginPath();
adaptiveCtx.arc(512, 512, 400, 0, Math.PI * 2);
adaptiveCtx.fill();

// Simple chart bars
const adaptiveBars = [
    { x: 400, y: 400, height: 150 },
    { x: 450, y: 350, height: 200 },
    { x: 500, y: 450, height: 100 },
    { x: 550, y: 380, height: 170 }
];

adaptiveCtx.fillStyle = '#1F2937';
adaptiveBars.forEach(bar => {
    adaptiveCtx.fillRect(bar.x, bar.y, 30, bar.height);
});

// SoroOne text
adaptiveCtx.fillStyle = '#1F2937';
adaptiveCtx.font = 'bold 80px Arial';
adaptiveCtx.textAlign = 'center';
adaptiveCtx.textBaseline = 'middle';
adaptiveCtx.fillText('SoroOne', 512, 650);

const adaptiveBuffer = adaptiveCanvas.toBuffer('image/png');
fs.writeFileSync('./assets/adaptive-icon.png', adaptiveBuffer);

console.log('SoroOne icons generated successfully!');

File: generate_icons.py
from PIL import Image, ImageDraw, ImageFont
import os

# Create a new image with a white gradient background
size = 1024
image = Image.new('RGBA', (size, size), color='white')
draw = ImageDraw.Draw(image)

# Create white to light gray gradient background
for y in range(size):
    # White to light gray gradient
    gray_value = int(255 - (7 * y/size))  # From 255 (white) to 248 (light gray)
    color = (gray_value, gray_value, gray_value, 255)
    draw.line([(0, y), (size, y)], fill=color)

# Draw chart container background
chart_bg = Image.new('RGBA', (size, size), (0, 0, 0, 0))
chart_draw = ImageDraw.Draw(chart_bg)
chart_draw.rounded_rectangle([150, 200, 874, 700], radius=20, fill=(255, 255, 255, 255), outline=(229, 231, 235, 255), width=4)
image = Image.alpha_composite(image, chart_bg)
draw = ImageDraw.Draw(image)

# Draw chart bars
bar_data = [
    (220, 350, 280),  # x, y, height
    (330, 280, 350),
    (440, 400, 230),
    (550, 320, 310),
    (660, 250, 380)
]

bar_color = (31, 41, 55)  # Dark gray bars

for x, y, height in bar_data:
    bar_img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    bar_draw = ImageDraw.Draw(bar_img)
    bar_draw.rounded_rectangle([x, y, x + 80, y + height], radius=8, fill=bar_color)
    image = Image.alpha_composite(image, bar_img)

# Draw trend line and dots
draw = ImageDraw.Draw(image)
trend_color = (59, 130, 246)  # Blue

# Draw line segments
line_points = [(260, 350), (370, 280), (480, 400), (590, 320), (700, 250)]
for i in range(len(line_points) - 1):
    draw.line([line_points[i], line_points[i+1]], fill=trend_color, width=6)

# Draw dots
for x, y in line_points:
    draw.ellipse([x-8, y-8, x+8, y+8], fill=trend_color)

# Draw axes
axis_color = (156, 163, 175)  # Gray
draw.line([(200, 650), (820, 650)], fill=axis_color, width=2)  # X-axis
draw.line([(200, 230), (200, 650)], fill=axis_color, width=2)  # Y-axis

# Draw the "SoroOne" text
try:
    # Try to use a system font
    font = ImageFont.truetype("arial.ttf", 120)
except:
    # Fallback to default font
    font = ImageFont.load_default()

# Get text dimensions
text = "SoroOne"
bbox = draw.textbbox((0, 0), text, font=font)
text_width = bbox[2] - bbox[0]
text_height = bbox[3] - bbox[1]

# Center the text
x = (size - text_width) // 2
y = 800 - text_height // 2

# Draw text shadow first
draw.text((x + 2, y + 2), text, font=font, fill=(0, 0, 0, 40))
# Draw main text
draw.text((x, y), text, font=font, fill=(31, 41, 55, 255))

# Save the icon
output_path = os.path.join(os.path.dirname(__file__), 'assets', 'icon.png')
image.save(output_path, 'PNG')

# Create adaptive icon (simpler version)
adaptive_image = Image.new('RGBA', (size, size), (0, 0, 0, 0))
adaptive_draw = ImageDraw.Draw(adaptive_image)

# Draw white circle background
adaptive_draw.ellipse([112, 112, size-112, size-112], fill=(255, 255, 255, 255))

# Draw simplified chart in center
mini_bars = [(400, 400, 150), (450, 350, 200), (500, 450, 100), (550, 380, 170)]
for x, y, height in mini_bars:
    adaptive_draw.rectangle([x, y, x + 30, y + height], fill=(31, 41, 55))

# Draw "SoroOne" text
try:
    adaptive_font = ImageFont.truetype("arial.ttf", 80)
except:
    adaptive_font = ImageFont.load_default()

# Center the "SoroOne" text at bottom
text = "SoroOne"
bbox = adaptive_draw.textbbox((0, 0), text, font=adaptive_font)
text_width = bbox[2] - bbox[0]
text_height = bbox[3] - bbox[1]
x = (size - text_width) // 2
y = 650

adaptive_draw.text((x, y), text, font=adaptive_font, fill=(31, 41, 55, 255))

# Save adaptive icon
adaptive_path = os.path.join(os.path.dirname(__file__), 'assets', 'adaptive-icon.png')
adaptive_image.save(adaptive_path, 'PNG')

print("Icons generated successfully!")
print(f"Main icon saved to: {output_path}")
print(f"Adaptive icon saved to: {adaptive_path}")
